pkg/minishift/config/validations.go

```diff
@@ -187,3 +187,35 @@ func IsValidNetmask(name string, mask string) error {
 
 	return nil
 }
+
+func IsValidPort(name string, p string) error {
+	port, err := strconv.Atoi(p)
+	if err != nil {
+		return fmt.Errorf("%s:%v", name, err)
+	}
+	if numInRange(port, 1024, 65536) {
+		if err := isPortAccessible(port); err != nil {
+			return fmt.Errorf("Port %d is not valid: %v", port, err)
+		}
+		return nil
+	} else {
+		return fmt.Errorf("Port %d is not valid", port)
+	}
+}
+
+func numInRange(num int, start int, end int) bool {
+	if num >= start && num <= end {
+		return true
+	}
+	return false
+}
+
+func isPortAccessible(port int) error {
+	hostWithPort := fmt.Sprintf(":%d", port)
+	lnr, err := net.Listen("tcp", hostWithPort)
```

Comment:
What is `lnr` instead of this can we use `conn` (sort form of connection)?

Reply:
Also does this block if port is blocked by firewalld, issue which @gbraad was facing that time?

Reply:
`lnr` was meant to be listener ;-), because it returns a listener. I guess if it is blocked by firewalld it will be able to still bind to the port, as host is localhost, but i'll verify this behavior.